store/sqlstore/channel_store.go

```diff
@@ -1755,13 +1756,74 @@ func (s SqlChannelStore) AutocompleteInTeamForSearch(teamId string, userId strin
 			}
 		}
 
+		directChannels, err := s.autocompleteInTeamForSearchDirectMessages(userId, term)
+		if err != nil {
+			result.Err = err
+			return
+		}
+
+		channels = append(channels, directChannels...)
+
 		sort.Slice(channels, func(a, b int) bool {
 			return strings.ToLower(channels[a].DisplayName) < strings.ToLower(channels[b].DisplayName)
 		})
 		result.Data = &channels
 	})
 }
 
+func (s SqlChannelStore) autocompleteInTeamForSearchDirectMessages(userId string, term string) ([]*model.Channel, *model.AppError) {
+	queryFormat := `
+			SELECT
+				C.*
+			FROM
+				Channels AS C
+			JOIN
+				ChannelMembers AS CM ON CM.ChannelId = C.Id
+			WHERE
+			    C.Type = 'D'
+				AND CM.UserId = :UserId
+				AND C.Id IN (
+					SELECT
+						ChannelId
+					FROM
+						Channels AS IC
+					JOIN
+						ChannelMembers AS ICM ON ICM.ChannelId=IC.Id
+					JOIN
+						Users AS IU ON ICM.UserId=IU.Id
+					WHERE
+						ChannelId = C.Id
+						%v
+				)
+				GROUP BY C.Id
```

Comment:
Can you explain the need for a `GROUP BY` here? There's no aggregation going on, and there should be no duplicate rows unless we allow duplicate `ChannelId, UserId` in the `ChannelMembers` table.

Reply:
Was because I rewrite the query some times, and at some time it can generate multiple channel lines derived from the join, and I wanted to avoid the duplications. Now I rewrite a lot of this and is working fine